### PR TITLE
Implemented new SFX including barks, updated MainMenu

### PR DESCRIPTION
**Pull request by gracebarsny**
**Resources**
- Added new SFX files (GiftRedeem, Back, Empty, Barks_Small, Barks_Medium, Barks_Large, Purchase)
- AudioList.json: updated to support new SFX
- GameData.json: updated to support breed sizes

**Scripts**
- AudioController, MonoBehaviourExtended, EventController: added handleNamedEventWithID to support a specific bark SFX with the same event call
- PPGameController: fixed currency bug, calls "PlayPurchase" event
- DogBreed: added field and accessor for size
- UIElement: marked Show, Hide, and Toggle as virtual
- DogBrowser, DogProfile: calls "PlayBack" before closing
- PPMainMenuUIController: added calls for PlayMenuClick, PlayMenuPopup, and PlayEmpty, added methods for loading the living room and yard scenes, overrides Toggle
- NavigationPanel: added calls to PlayMenuClick
- GiftReportUI: overrides Destroy, calls PlayBack and PlayGiftRedeem
- SettingsPopup: overrides Hide, calls PlayBack
- PPShelterUIController: calls PlayBark
- AudioUtil: added HasId methods

### CODE REVIEW

File: Assets/Scripts/Controller/PickupPup/PPGameController.cs [hunk 1]
```diff
@@ -298,9 +298,9 @@ public CurrencyData GetGift(DogDescriptor dog)
     public bool TryBuyItem(int value, CurrencyType valueCurrencyType,
         int cost, CurrencyType costCurrencyType)
     {
-        if (CanAfford(costCurrencyType, cost))
+        if (!CanAfford(costCurrencyType, cost))
         {
-			print("Failed To Buy");
+            EventController.Event("PlayEmpty");
```
Comment (imann24) [inline Assets/Scripts/Controller/PickupPup/PPGameController.cs]: I think these constants should be refactored to be stored inside of PPGlobal.cs

File: Assets/Scripts/Controller/PickupPup/PPGameController.cs [hunk 2]
```diff
@@ -315,6 +315,7 @@ public bool TryBuyItem(ShopItem item)
     void buyItem(int value, CurrencyType valueCurrencyType,
         int cost, CurrencyType costCurrencyType)
     {
+        EventController.Event("PlayPurchase");
```
Comment (imann24) [inline Assets/Scripts/Controller/PickupPup/PPGameController.cs]: Note about constants applies to all hardcoded audio string events

File: Assets/Scripts/UI/PickupPup/DogBrowser/DogBrowser.cs [hunk 3]
```diff
@@ -81,6 +81,7 @@ public void Open(int pageIndex = NONE_VALUE)
 
 	public void Close()
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogBrowser/DogBrowser.cs]: I think I wrote this, but on review, I don't see why this isn't just an override of the `Hide` method. Seems cleaner

File: Assets/Scripts/Util/AudioUtil.cs [hunk 4]
```diff
@@ -5,9 +5,13 @@
 
 using UnityEngine;
 using System.Collections;
+using System.Text.RegularExpressions;
+using k = Global;
 
 public static class AudioUtil 
 {
+    static Regex idRgx = new Regex(".+" + k.JOIN_CHAR + ".+");
```
Comment (imann24) [inline Assets/Scripts/Util/AudioUtil.cs]: I don't know enough about Regex to properly comment, but consider moving these substrings into Global

Comment (gracebarsny) [inline Assets/Scripts/Util/AudioUtil.cs]: Regex can't be constants. But I should still move the substring?

Comment (gracebarsny) [inline Assets/Scripts/Util/AudioUtil.cs]: So moving the substring is a mess.

Comment (imann24) [inline Assets/Scripts/Util/AudioUtil.cs]: Cool to leave as is then

Comment (gracebarsny) [inline Assets/Scripts/Util/AudioUtil.cs]: Thank you.

**Pull request merged**